R/waitUntilFinished.R

```diff
@@ -38,16 +38,21 @@ setMethod(f = "wait",signature(.Object = "character", sleep.time = "numeric"), d
   running <- running(.Object, retry = TRUE)
   pb <- progress_bar$new(total = 100, clear = FALSE)
   currentStatus <- 'unknown'
+  succeededYet <- FALSE
   while(running){
     Sys.sleep(sleep.time)
     running <- running(.Object, retry = TRUE)
     checkResult <- check(.Object)
-    if(!is.null(checkResult$percentComplete)) {
-      pb$update(as.numeric(checkResult$percentComplete)/100)
+    percentComplete <- checkResult$percentComplete
+    if(!is.null(percentComplete) && !succeededYet) {
+      pb$update(as.numeric(percentComplete)/100)
     } else if(checkResult$status != currentStatus) {
       message(checkResult$status)
     }
     currentStatus <- checkResult$status
+    if(checkResult$statusType == "ProcessSucceeded" && percentComplete == 100) {
```

Comment:
Haven't tested that myself explicitly, but that shouldn't cause any problems.  In that case the progress bar should just not start I think, or worst case would not visually complete.  The progress bar code here doesn't impact the actual `while` loop here though, so it shouldn't affect the actual outcome of the job. 

As far as addressing this in `running`, it would be good to have an opinion from someone with better knowledge of the inner workings of the GDP code base.  My feeling though is that the percent complete number isn't rock-solid reliable, so we are best off to only use it in the progress bar, which is ultimately a cosmetic feature, rather than using it to tell when a job is completed elsewhere. Based on Gretchen's examples, it seems like on big jobs there is some lag between the `percentComplete` reading 100% and the job actually finishing (perhaps the output file is writing to disk or something?).  So it is maybe safer to stick with using the status as the indicator for the job completing.

Reply:
Ok, understood - good point(s). 

Given the potential collision with progress bars in other packages, I also wonder if it is worth having an argument `...,progress = TRUE` that is also something you can set in `gconfig()`. That way you can turn off the progress bar when it is unnecessary or would collide with other progress bars (e.g., `scipiper`'s loop_tasks) 

I think that would be a different issue we should create, but how isolated is the `progress` code? is it all within this one function?

Reply:
I agree that would be useful.  This function has the only `progress` code, so it shouldn't be hard to add.